### ISSUE TITLE
Autoimported object has too many common names in scope

### ISSUE THREAD

**Issue by jvican** [agemooij/sbt-prompt#18]
`autoImport` extends several traits that define lots of common operations like `currentSbtVersion`, `currentScalaVersion`, etcetera. https://github.com/agemooij/sbt-prompt/blob/master/src/main/scala/com/scalapenos/sbt/prompt/promptlets/promptlets.scala#L16-L36
  
This causes name clash problems with builds that *do* define these names in some way or another. In my case, it failed because I had a `val currentScalaVersion = "2.12.3"` in my `build.sbt`.
  
This is an important problem. Sbt autoimports all those names and that's why it's good practice to reduce the names in `autoImport` as much as possible.
  
Would it be possible that you remove all those clashing names from this excellent sbt plugin?

**Comment by agemooij** [comment]
Wow, good point! I never thought about that at the time and in the mean time I've actually learned to use `autoImport` better.

I'll try to fix this ASAP. For extra speed it would be great if you could send a PR.

**Comment by agemooij** [comment]
Reading up on how I did this, I now remember why it ended up like this. 
I wanted all the DSL building block functions, like `text` and `bg`, to be auto imported so people could call them to construct prompts without needing an explicit import. The fact that these functions clash with general SBT settings is just me being an idiot and not considering the impact.

The obvious solutions would be:

- rename the relevant functions so they don't clash, perhaps using a prefix
- or to remove these functions from the top-level `autoImport` and instead expose them nested under a `Promptlets` object for namespacing.

I like the second option best. Would you agree @jvican ?

**Comment by jvican** [comment]
I think the second option is best @agemooij, it addresses the issue at the root of the problem. The fewer names in `autoImport`, the better :smile:. That is a breaking change, it would probably deserve a `2.0` release.

**Comment by agemooij** [comment]
I agree. I'll see if I can find some time to implement this. A PR would certainly help speed things up 😉

**Comment by jvican** [comment]
Would love to help, but I don't have time.
